app/controllers/api/v0/events_controller.rb

```diff
@@ -16,23 +16,59 @@ def create
     render nothing: true, status: 201
   end
 
-  private
+  class KafkaData
+    def initialize(api_data:, controller:)
+      @data = api_data.to_hash
```

Comment:
Let's clone the `api_data` so we don't leak changes from here outside of this class.  `to_hash` doesn't clone:

```bash
irb(main):002:0> x = {foo: 2}
=> {:foo=>2}
irb(main):003:0> y = x.to_hash
=> {:foo=>2}
irb(main):004:0> y[:foo] = 3
=> 3
irb(main):005:0> x
=> {:foo=>3}
```

Cloning:

```bash
irb(main):006:0> x = {foo: 2}
=> {:foo=>2}
irb(main):007:0> y = x.clone
=> {:foo=>2}
irb(main):008:0> y[:foo] = 3
=> 3
irb(main):009:0> x
=> {:foo=>2}
```

Maybe `#deep_dup` is available?

Reply:
done